agent/algorithm/python/python.go

```diff
@@ -59,17 +54,13 @@ func New(logger *slog.Logger, eventsSvc events.Service, runtime, requirementsFil
 	return p
 }
 
-func (p *python) AddDataset(dataset string) {
-	p.datasets = append(p.datasets, dataset)
-}
-
-func (p *python) Run() ([]byte, error) {
+func (p *python) Run(withDataset bool) error {
```

Comment:
we don't need this condition or argument to algo, datasets dir is immutable and we don't need to pass it to algo